src/app/modules/spcp/spcp.service.ts

```diff
@@ -152,4 +153,43 @@ export class SpcpService {
       return ok({ isValid: false, errorCode })
     }
   }
+
+  extractJwtPayload(
+    jwt: string,
+    authType: AuthType.SP | AuthType.CP,
+  ): ResultAsync<JwtPayload, VerifyJwtError | InvalidAuthTypeError> {
+    let authClient: SPCPAuthClient
+    switch (authType) {
+      case AuthType.SP:
+        authClient = this.#singpassAuthClient
+        break
+      case AuthType.CP:
+        authClient = this.#corppassAuthClient
+        break
+      default:
+        return errAsync(new InvalidAuthTypeError(authType))
+    }
+    const payloadPromise = new Promise<JwtPayload>((resolve, reject) => {
+      authClient.verifyJWT<JwtPayload>(
+        jwt,
+        (error: Error, data: JwtPayload) => {
+          if (error) {
+            return reject(error)
+          }
+          return resolve(data)
+        },
+      )
+    })
```

Comment:
i think the code will be cleaner if promisification is performed outside of the service function.